### ISSUE TITLE
Glance image cache should be turned off

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-openstack#1005]
##### What version (commit) do you encounter this issue on?

All

##### What type of issue is this (bug, feature, documentation, regression)?

Feature

##### What is the expected behavior?

No  image create from a cinder volume should be failing due to a small glance-api container  root disk

##### What is actually happening?

The glance-api containers use a too small root disk for a lot of image conversions and cache related activities. Hence any image with exceeds the initial 10G will fail during conversions etc

##### How can we reproduce this issue?

Make a glance image from a 40G cinder volume

##### How can we verify your issue is addressed?

Images  in the size of the available capacity of /openstack  are supported 

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?

Minor

##### What is your evaluation on the importance of this issue, and why?

- 

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.1
- [ ] liberty-12.0
- [ ] kilo

**Comment by odyssey4me** [comment]
I very strongly disagree that the image cache should be turned off as a rule. The image cache is there to improve the end-user experience when using networked image stores (eg Cloudfiles). It delivers the image from the local cache instead of having to stream the image from the store for every request coming from a new compute node.

The facility is already provided to set the size of the Glance container for the environment. I would recommend that the size be set based on the needs of the environment to cater for the expected image cache size.

An alternative is to place the cache on the host and bind-mount it into the container. I don't think this is a good idea as it could result in the customer doing something which results in the /openstack directory filling up and causing database corruption. As I understand it the /openstack directory is a shared data store between the database, logs and backups. If this is used, then a special partition should be implemented to bind-mount it into.

**Comment by BjoernT** [comment]
FYI we turn it mostly off on all deployments using large images. In real live customer have > 120GB images and extending the container is not really an option. Currently we see day to day corrupt images in the cache or non working container. The container is always too small and I do not want to run a glance image container or a cinder volumes container to 1TB. Also the images are stored locally on the nova compute nodes so the customers didn't come back with negative experience yet.

**Comment by mancdaz** [comment]
@BjoernT @odyssey4me I guess it comes down to a question of how often this gets disabled in rpc deployments anyway. If it is already being disabled by support on >50% of deployments by overriding the var (or worse, by hand editing files) per customer, then the default in rpc should probably change.

if not, then the default should leave it enabled and individual customers have it turned off on an as-needed basis.

Point being, the discussion here is not about the validity of doing so in any particular deployment - we're simply talking about changing the default to the thing used most often. Whether we do or do not change the default here, it can and will still be overridden at deployment time if the decision is made for an individual customer.

As far as the related PR, I have asked it gets split into two - it is attempting to address two separate issues in a single commit, which is against our contributing guidelines.

**Comment by BjoernT** [comment]
So let's see it from this perspective, I want to fix an issue before we even get a ticket from the customer and someone else has to deal with it again. A change of the default value is necessary this is out of question, our container images are just too small. I'm happy to change the cinder portion since we can assume that cinder is deployed on bare metal and we can create a /openstack/cinder_cache or so  and use that directory and no bind mount would be necessary

**Comment by mancdaz** [comment]
@BjoernT yeah I guess what I'm saying is that as consumers of this stuff, if you want a default to be set one way or another then we can do that - it's an operational decision that you are making on behalf of the support team and customers so I'm ok with that in principle.

But yes you'll need to remove the cinder bits from the PR and leave the PR targeted at that single change, and I will then +1. You'll then need to find another +1 :)

**Comment by BjoernT** [comment]
Just uploaded a new patch which is also rebased now, github seems to lag a little bit

**Comment by BjoernT** [comment]
@alextricity25  This  has merged in master and I like to backport it into mitaka
